rosidl_default_generators/rosidl_default_generators-extras.cmake.in

```diff
@@ -0,0 +1,29 @@
+# generated from rosidl_default_generators/rosidl_default_generators-extras.cmake.in
+
+set(_exported_dependencies
+  "rosidl_typesupport_introspection_cpp"
+  "rosidl_typesupport_connext_cpp"
+  "rosidl_typesupport_opensplice_cpp"
+)
+
+# find_package() all dependencies (if available)
+# and append their DEFINITIONS INCLUDE_DIRS and LIBRARIES variables
+# to @PROJECT_NAME@_DEFINITIONS , @PROJECT_NAME@_INCLUDE_DIRS and
+# @PROJECT_NAME@_LIBRARIES.
+# Additionally collect the direct dependency names in
+# @PROJECT_NAME@_DEPENDENCIES as well as the recursive dependency names
+# in @PROJECT_NAME@_RECURSIVE_DEPENDENCIES.
+foreach(_dep ${_exported_dependencies})
+  find_package("${_dep}" QUIET)
+  if(${_dep}_FOUND)
+    if(${_dep}_DEFINITIONS)
+      list(APPEND @PROJECT_NAME@_DEFINITIONS "${${_dep}_DEFINITIONS}")
+    endif()
+    if(${_dep}_INCLUDE_DIRS)
+      list(APPEND @PROJECT_NAME@_INCLUDE_DIRS "${${_dep}_INCLUDE_DIRS}")
+    endif()
+    if(${_dep}_LIBRARIES)
+      list(APPEND @PROJECT_NAME@_LIBRARIES "${${_dep}_LIBRARIES}")
+    endif()
+  endif()
+endforeach()
```

Comment:
This code looks very similar to https://github.com/ament/ament_cmake/blob/master/ament_cmake_export_dependencies/cmake/ament_cmake_export_dependencies-extras.cmake.in

Maybe it should be refactored so that the logic can be reused here. I also noticed that the comment for this foreach loop mentions the `@PROJECT_NAME@_RECURSIVE_DEPENDENCIES` variable, but doesn't actually create it.